client/pages/index/index.js

```diff
@@ -8,69 +10,132 @@ Page({
-   * 生命周期函数--监听页面隐藏
-   */
-  onHide: function () {
-
-  },
-
-  /**
-   * 生命周期函数--监听页面卸载
-   */
-  onUnload: function () {
-
-  },
-
-  /**
-   * 页面相关事件处理函数--监听用户下拉动作
-   */
-  onPullDownRefresh: function () {
-
-  },
-
-  /**
-   * 页面上拉触底事件的处理函数
-   */
-  onReachBottom: function () {
-
-  },
-
-  /**
-   * 用户点击右上角分享
-   */
-  onShareAppMessage: function () {
-
+  onShow: function(options) {
+    var that = this;
+    qcloud.request({
+      // url: 'http://localhost:5757/weapp/user',
+      url: 'https://aeoo5f7q.qcloud.la/weapp/user',
+      method: 'GET',
+      header: {
+        'content-type': 'application/x-www-form-urlencoded'
+      },
+      data: {
+        OpenId: 1234,
+        //'1234'should be the user's openid, here is just an demo. One admin with OpenId =1234 was written into local DB for test purpose.
+      },
+      success: function(res) {
+        console.log(res);
+        if (res.data.IsAdmin == 1) {
+          that.setData({
+            hid: false
+          })
+        }
+      }
+    });
+    qcloud.request({
+      // url: 'http://localhost:5757/weapp/training',
+      url: 'https://aeoo5f7q.qcloud.la/weapp/training',      
+      method: 'GET',
+      header: {
+        'content-type': 'application/x-www-form-urlencoded'
+      },
+      success: function(res) {
+        console.log(res);
+        var data = res.data[0];
+        var rowTime = data.Time;
+        var newTime = rowTime.slice(0, 5);
+        var dataAndTime = [data.Day, newTime].join(' ');
+        that.setData({
+          location: data.Location,
+          detailed: data.LocationDetail,
+          time: dataAndTime
+        })
+        const weekday = [
+          "Monday",
+          "Tuesday",
+          "Wednesday",
+          "Thursday",
+          "Friday",
+          "Saturday",
+          "Sunday"
+        ];
+        var editDay = new Date(data.UpdatedTime).getDay();
+        var trainingDay = weekday.indexOf(data.Day) + 1;
+        var trainingDate = new Date(data.UpdatedTime);
+        trainingDate.setDate(trainingDate.getDate() + trainingDay - editDay);
+        trainingDate.setHours(parseInt(rowTime.slice(0, 3)), parseInt(rowTime.slice(3, 5)));
+        var currentDate = new Date();
+        if (currentDate > trainingDate) {
+          that.setData({
+            edit: 'Add',
+          })
+        } else {
+          that.setData({
+            edit: 'Edit',
+          })
+        }
+        const usualLocation = {
+          'NUS': {
+            lat: 1.29877,
+            lon: 103.77821
+          },
+          'NTU': {
+            lat: 1.348934,
+            lon: 103.688587
+          },
+          'Nan Hua': {
+            lat: 1.308768,
+            lon: 103.769796
+          }
+        }
+        let lat = 1.29877;
+        let lon = 103.77821 //Default NUS
+        if (['NTU', 'Nan Hua'].includes(data.Location)){
+          lat = usualLocation[data.Location].lat;
+          lon = usualLocation[data.Location].lon;
+        }
```

Comment:
can we set the default values of lat and lon to the lat and lon of Singapore, and if usualLocations include data.Location, set lat and lon accordingly.